src/java.base/share/classes/java/util/jar/JarInputStream.java

```diff
@@ -90,7 +90,7 @@ private JarEntry checkManifest(JarEntry e)
     {
         if (e != null && JarFile.MANIFEST_NAME.equalsIgnoreCase(e.getName())) {
             man = new Manifest();
-            byte bytes[] = getBytes(new BufferedInputStream(this));
+            byte[] bytes = new BufferedInputStream(this).readAllBytes();
```

Comment:
I wonder if it makes sense to avoid reading the entire manifest into a byte[] when we don't need to verify the JAR. This may help avoiding some intermediate allocation and copying. This make be noticeable for some of the larger manifests (Java EE, OSGi, ...). A possible implementation may look like this https://github.com/marschall/jdk/commit/c50880ffb18607077c4da3456b27957d1df8edb7.

In either case since we're calling #readAllBytes I don't see why we are wrapping in a BufferedInputStream rather than calling #readAllBytes directly.

Reply:
Usage of `BufferedInputStream` removed